miner/block_builder.go

```diff
@@ -464,24 +378,26 @@ func (t *BlockBuilder) acceptBlockData() {
 			}
 			// TODO: include multiple proofs in each block and weigh blocks where applicable
 
-			var atxList []types.ATXID
-			for _, atx := range t.AtxPool.GetAllItems() {
-				atxList = append(atxList, atx.ID())
-			}
-
+			//reducedAtxList := selectAtxs(atxList, t.atxsPerBlock)
 			for _, eligibilityProof := range proofs {
-				txList, err := t.TransactionPool.GetTxsForBlock(t.txsPerBlock, t.projector.GetProjection)
+				txList, txs, err := t.TransactionPool.GetTxsForBlock(t.txsPerBlock, t.projector.GetProjection)
 				if err != nil {
 					events.Publish(events.DoneCreatingBlock{Eligible: true, Layer: uint64(layerID), Error: "failed to get txs for block"})
 					t.With().Error("failed to get txs for block", layerID, log.Err(err))
 					continue
 				}
-				blk, err := t.createBlock(layerID, atxID, eligibilityProof, txList, atxList)
+				blk, err := t.createBlock(layerID, atxID, eligibilityProof, txList)
 				if err != nil {
 					events.Publish(events.DoneCreatingBlock{Eligible: true, Layer: uint64(layerID), Error: "cannot create new block"})
 					t.Error("cannot create new block, %v ", err)
 					continue
 				}
+				err = t.meshProvider.AddBlockWithTxs(blk, txs, nil)
```

Comment:
keep in mind saving to db happens again when block is received by our own gossip. (maybe we should add a special case?)

Reply:
it doesn't we have a check for that in gossip